src/Propel/Generator/Model/Behavior.php

```diff
@@ -169,6 +169,24 @@ public function getTable(): ?Table
         return $this->table;
     }
 
+    /**
+     * Returns the table this behavior is applied to
+     *
+     * @throws \Propel\Generator\Exception\LogicException
+     *
+     * @return \Propel\Generator\Model\Table
+     */
+    public function getTableOrFail(): Table
```

Comment:
@dmytro-dymarchuk

Reply:
Since the property doesn't have a default value and is not initialized we can't be sure that the property is not null.

Reply:
Oh, you misunderstood my question. I am aware that the way this is implemented, the property could be null, but I was wondering, since the method is never used internally, which use case this supports. But I understand now, the answer is, that when users write their own behaviors, they extend this class and the new method allows them to avoid type errors.
I am still not sure if this is similarly true for all the new `getXYZorFail` methods, because I suspect that some of the classes cannot be used in userland, like the Generator classes. Which would mean that a method that is not used internally will never be used an can be omitted. But since no one has time to investigate, I guess this doesn't matter.

Reply:
I investigated the possibility to avoid creating `getXYZorFail` and did it by changing the constructor where it is possible without massive refactoring.

Reply:
Right, yes, sorry, I only saw it after I posted the comment. Thank you!